src/mscorlib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -800,5 +801,83 @@ public override Module[] GetLoadedModules(bool getResourceModules)
 
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         internal static extern int GetToken(RuntimeAssembly assembly);
+
+        public sealed override Type[] GetForwardedTypes()
+        {
+            List<Type> types = new List<Type>();
+            List<Exception> exceptions = new List<Exception>();
+
+            MetadataImport scope = GetManifestModule(GetNativeHandle()).MetadataImport;
+            scope.Enum(MetadataTokenType.ExportedType, 0, out MetadataEnumResult enumResult);
+            for (int i = 0; i < enumResult.Length; i++)
+            {
+                MetadataToken mdtExternalType = enumResult[i];
+                Type type = null;
+                Exception exception = null;
+                ObjectHandleOnStack pType = JitHelpers.GetObjectHandleOnStack(ref type);
+                try
+                {
+                    GetForwardedType(this, mdtExternalType, pType);
+                    if (type == null)
+                        continue;  // mdtExternalType was not a forwarder entry.
+                }
+                catch (Exception e) when (IsLoaderException(e))
```

Comment:
Yes, it's meant to mimic the CoreRT implementation where the type loading primitives reports all loader-related exceptions in a non-throwing manner.